tests/test_s3.py

```diff
@@ -114,6 +115,107 @@ def test_metrics(self):
             'AWS/S3.NumberOfObjects.Average' in resources[0]['c7n.metrics'])
 
 
+class AccessFilter(BaseTest):
+
+    username = 'test-user-for-access-filter'
+    user_arn = 'arn:aws:iam::114002464496:user/' + username
+
+    @classmethod
+    def setUpClass(cls):
+        if not C7N_RECORD:
+            return
+        cls._bare_iam = boto3.Session().client('iam')
+        response = cls._bare_iam.create_user(UserName=cls.username)
+        assert cls.user_arn == response['User']['Arn']
+
+    @classmethod
+    def tearDownClass(cls):
+        if not C7N_RECORD:
+            return
+        cls._bare_iam.delete_user(UserName=cls.username)
+
+
+    def check(self, flight, result_from_iam, result_from_resource):
+        bname = 'test-bucket-for-access-filter'
+        self.patch(s3.S3, 'executor_factory', MainThreadExecutor)
+        self.patch(s3, 'S3_AUGMENT_TABLE', [
+            ('get_bucket_policy',  'Policy', None, 'Policy'),
+        ])
+        session_factory = self.get_session_factory(flight)
+        session = session_factory()
+
+        # create a bucket
+        client = session.client('s3')
+        client.create_bucket(Bucket=bname)
+        self.addCleanup(destroyBucket, client, bname)
+
+        # create a user with certain bucket perms
+        iam = session.client('iam')
+        bucket_arn = "arn:aws:s3:::%s/*" % bname
+        if result_from_iam is not None:
+            effect = 'Allow' if result_from_iam else 'Deny'
+            iam.put_user_policy(
+                UserName=self.username,
+                PolicyName='test-policy',
+                PolicyDocument='''{"Version": "2012-10-17", "Statement":[{
+                    "Effect": "%s",
+                    "Resource": "%s",
+                    "Action": ["s3:*"]}]}
+                ''' % (effect, bucket_arn))
+        if result_from_resource is not None:
+            if C7N_RECORD:
```

Comment:
I didn't find one, so I added `.recording` in d5be20a96e28b34125e72c1e16198c92957df392. Let me know if I'm missing something ...